utils/event_core.lua

```diff
@@ -71,7 +85,15 @@ end
 
 local function on_nth_tick_event(event)
     local handlers = on_nth_tick_event_handlers[event.nth_tick]
+    local profiler = on_nth_tick_event_profilers[event.nth_tick]
+    if not profiler then
+        profiler = game.create_profiler()
+        on_nth_tick_event_profilers[event.nth_tick] = profiler
+    else
+        profiler.restart()
+    end
     call_handlers(handlers, event)
+    profiler.stop()
```

Comment:
Could simplify and remove duplication by creating a shared&reusable utils.profile block.
```suggestion
    profile(on_nth_tick_event_profilers, event.nth_tick, function()
      call_handlers(handlers, event)
    end)
```

Don't think it matters tho. Could be overhead from closure, but probably negligible difference.